test/e2e/test_cap_kill.py

```diff
@@ -140,3 +142,177 @@ def test_cap_kill_not_inherited_by_running_jobs(
+        deadline_client=deadline_client,
+        logs_client=logs_client,
+        expected_pattern=re.escape("Ambient set =\n"),
+    )
+
+
+@pytest.mark.skipif(
+    os.environ["OPERATING_SYSTEM"] == "windows",
+    reason="Linux specific test",
+)
+@pytest.mark.usefixtures("session_worker")
+def test_worker_only_has_cap_kill(
+    deadline_client: DeadlineClient,
+    deadline_resources: DeadlineResources,
+) -> None:
+    """Tests to make sure that the worker agent has no capabilities besides cap_kill"""
+    job: Job = Job.submit(
+        client=deadline_client,
+        farm=deadline_resources.farm,
+        queue=deadline_resources.queue_a,
+        priority=98,
+        max_retries_per_task=1,
+        template={
+            "specificationVersion": "jobtemplate-2023-09",
+            "name": "Check worker process Linux capabilities",
+            "description": "Test that worker only has cap_kill",
+            "steps": [
+                {
+                    "hostRequirements": {
+                        "attributes": [
+                            {
+                                "name": "attr.worker.os.family",
+                                "allOf": [os.environ["OPERATING_SYSTEM"]],
+                            }
+                        ]
+                    },
+                    "name": "Step0",
+                    "script": {
+                        "actions": {
+                            "onRun": {
+                                "command": "bash",
+                                "args": ["{{Task.File.CheckCaps}}"],
+                            },
+                        },
+                        "embeddedFiles": [
+                            {
+                                "name": "CheckCaps",
+                                "type": "TEXT",
+                                "runnable": True,
+                                # The worker process is the parent of the parent of the current process
+                                # command substitutions (eg. `$()`) run in a new process, so we need to save `$PPID`
+                                # which is a special variable that refers to the parent PID of the current shell, then use
+                                # `ps` to get the parent PID of that, which will be the worker's PID. We then call `getpcaps`
+                                # on that PID to get the worker's capabilities. The output for a process without any capabilities
+                                # will look like:
+                                #
+                                # ```
+                                # <pid>: =ep
+                                # ```
+                                # whereas the output for a process that only has cap_kill (what the worker is supposed to have)
+                                # will look like this:
+                                # ```
+                                # <pid>: cap_kill=ep
+                                # ```
```

Comment:
We should state that this is the expected output when `CAP_KILL` is in the effective and permitted capability sets of the worker agent process. We specifically don't want to see `cap_kill=eip` which would mean the agent has `CAP_KILL` in its inheritable capability set which would result in subprocesses having this capability.